utils/loading_cache.hh

```diff
@@ -290,6 +290,31 @@ public:
         });
     }
 
+    void put(const Key& k, value_type value) {
+        // We shouldn't be here if caching is disabled
+        assert(caching_enabled());
+
+        timestamped_val_ptr ts_val_ptr = _loading_values.get_or_insert(k, std::move(value));
+        // check again since it could have already been inserted and initialized
+        if (!ts_val_ptr->ready() && !ts_val_ptr.orphaned()) {
+            _logger.trace("{}: putting the value for the first time", k);
+
+            if (ts_val_ptr->size() > _max_size) {
+                throw entry_is_too_big();
```

Comment:
What would the caller do with this exception?

In general caches should try to be transparent and not burden the caller with exceptions they can't handle.

Reply:
I assumed that if the caller can handle the exception(future) in get_ptr, he can also handle an exception here. 
To handle this, the caller can for example give up and expect that the next user will load the value himself, or try again with a smaller entry if he has an access to one

Reply:
Ah, I see it's an existing problem. Well, let's not solve it here.